lib/fizzy/stack.hpp

```diff
@@ -61,37 +67,63 @@ class OperandStack
     /// in the constructor after the m_storage.
     Value* m_top;
 
+    Value* m_bottom;
+
+    Value* m_locals;
+
     /// The pre-allocated internal storage.
     Value m_small_storage[small_storage_size];
 
     /// The unbounded storage for items.
     std::unique_ptr<Value[]> m_large_storage;
 
-    Value* bottom() noexcept { return m_large_storage ? m_large_storage.get() : m_small_storage; }
-
-    const Value* bottom() const noexcept
-    {
-        return m_large_storage ? m_large_storage.get() : m_small_storage;
-    }
-
 public:
     /// Default constructor.
     ///
-    /// Based on @p max_stack_height decides if to use small pre-allocated storage or allocate
-    /// large storage.
-    /// Sets the top item pointer to below the stack bottom.
-    explicit OperandStack(size_t max_stack_height)
+    /// Based on required storage space decides to use small pre-allocated
+    /// storage or allocate large storage.
+    /// Sets the top stack operand pointer to below the operand stack bottom.
+    /// @param args                 Function arguments. Values are copied at the beginning of the
+    /// storage space.
+    /// @param num_local_variables  The number of the function local variables (excluding
+    ///                             arguments). This number of values is zeroed in the storage space
+    ///                             after the arguments.
+    /// @param max_stack_height     The maximum operand stack height in the function. This excludes args and num_local_variables.
+    OperandStack(span<const Value> args, size_t num_local_variables, size_t max_stack_height)
     {
-        if (max_stack_height > small_storage_size)
-            m_large_storage = std::make_unique<Value[]>(max_stack_height);
-        m_top = bottom() - 1;
+        const auto num_args = args.size();
```

Comment:
Perhaps this one could make sense to be typed as `size_t` because then we can be sure `storage_size_required` is `size_t`?  Though I guess `size()` returns that anyhow.

Reply:
This is fine now, but probably should be changed to `uint32_t` at some point.